test/unit/indexer/compare_enumerator_test.rb

```diff
@@ -0,0 +1,133 @@
+require "test_helper"
+require 'indexer/compare_enumerator'
+
+class CompareEnumeratorTest < MiniTest::Unit::TestCase
+  def test_when_matching_keys_exist
```

Comment:
I think it'd also be handy to have a test with a bit more data in each enumerator so that we test what happens _after_ the enumerator encounters a missing value, i.e. does it correctly increment just one the indexes?